### ISSUE TITLE
Error when file has gzip on

### ISSUE THREAD

**Issue by ivankennethwang** [daichirata/gcsproxy#6]
Can't fetch file properly if the file has gzip on in metadata

**Comment by gWOLF3** [comment]
@ivankennethwang were you able to solve this issue or find a workaround? 

i am having the same problem, and have tried playing around with all the nginx gzip config options..

http://nginx.org/en/docs/http/ngx_http_gzip_module.html

**Comment by gWOLF3** [comment]
@ivankennethwang - Hey bro I fixed this, will submit a PR. The appended content encoding in the proxy response should be based on the res that the storage reader gets, rather than the api response from content encoding in bucket, to handle cases where content in decompressed from server. 

The fix was quite simple once I understood what was happening. See my new proxy function as an example:
```
func proxy(w http.ResponseWriter, r *http.Request) {
  params := mux.Vars(r)
  obj := client.Bucket(params["bucket"]).Object(params["object"]).ReadCompressed(false)
  objr, err := obj.NewReader(ctx)
  attrC := objr.Attrs.ContentEncoding
  attr, err := obj.Attrs(ctx)
  fmt.Printf("CONTENT-ENCODING%+v\n", attrC)
  //fmt.Printf("%+v\n", obj)
  if err != nil {
    log.Fatal(err)
    handleError(w, err)
    return
  }
  fmt.Printf("%+v\n", attr)
  setStrHeader(w, "Content-Type", attr.ContentType)
  setStrHeader(w, "Content-Language", attr.ContentLanguage)
  setStrHeader(w, "Cache-Control", attr.CacheControl)
  //setStrHeader(w, "Content-Encoding", objr.Attrs.ContentEncoding)
  setStrHeader(w, "Content-Encoding", attrC)
  setStrHeader(w, "Content-Disposition", attr.ContentDisposition)
  if err != nil {
    log.Fatal(err)
    handleError(w, err)
    return
  }
  io.Copy(w, objr)
}


```

**Comment by daichirata** [comment]
I believe this issue has been resolved by #10 and #16. Thanks to all who helped.
